example/pythonlib/module/1-common-config/build.mill

```diff
@@ -0,0 +1,99 @@
+// This example shows some of the common tasks you may want to override on a
+// `PythonModule`: specifying the `mainScript`, adding additional
+// sources/resources, generating sources, and setting typecheck/run options.
+// Also we have support for `forkenv` and  `pythonOptions` to allow user to
+// add variables for environment and options for python respectively.
+
+package build
+import mill._, pythonlib._
+
+object foo extends PythonModule {
+  // You can have arbitrary numbers of third-party libraries
+  def pythonDeps = Seq("MarkupSafe==3.0.2", "Jinja2==3.1.4")
+
+  // choose a main Script to run if there are multiple present
+  def mainScript = Task.Source { millSourcePath / "custom-src" / "foo2.py" }
+
+  // TODO: should we include only one(it is sufficient) or both tasks(for demo purpose)?
+  // Add (or replace) source folders for the module to use
+  def sources = Task.Sources {
+    super.sources() ++ Seq(PathRef(millSourcePath))
+  }
+
+  // Add (or replace) resource folders for the module to use
+  def resources = Task.Sources {
+    super.resources() ++ Seq(PathRef(millSourcePath))
```

Comment:
I would suggest not using `millSourcePath` in this example, since it has the potential to lead to overlapping PYTHONPATHS, which can be confusing. For example, as it currently stands, we can access `MyResource.txt` at the root of a pythonpath, by `MyOtherResources.txt` lives under the python module `custom-resources`.

For simplicity, I would suggest using a path at the same level as the existing `resouces/` path. Perhaps `custom-resources`, and you nest `MyOtherResoucre` in another folder.

Reply:
i have used one approach before where i am adding the `resources` and `custom-src` and others in the PYTHONPATH(means in their tasks) and then add a **`src`**  folder to give nest feel this fix the problem 

is this acceptable ?

Reply:
If you put `src` in resources, and `resources` is in the `PYTHONPATH`, then `src` becomes a python module (which is more or less the same as a java package). That means, that if you want to load the resource with import lib, you need to call `importlib.resources.read_text("src", "data.txt")`. Instead of using `src`, why not reuse the same module you already have, `foo` in this case?

Reply:
i tried `foo.resources` and `foo.custom-resources` (also with `resources` and `custom-resources`)and it said ModuleNotFoundError: No module named 'foo.resources'; 'foo' is not a package

This means we have to add `millSourcePath` to the PYTHONPATH for making it's subdirectories as Module and this bring us to again same thing as we have now

Reply:
@jodersky Sir let's do resources/ res and custom-resources/ custom-res 

is this fine ?

Reply:
> i tried foo.resources and foo.custom-resources (also with resources and custom-resources)and it said ModuleNotFoundError: No module named 'foo.resources'; 'foo' is not a package

You wouldn't need `foo.resources`. Essentially, if PYTHONPATH includes `/a/b/c`, and your resource is in `/a/b/c/x/y/z`, then you would access it in importlib with `x.y.z`.

Anyway, I think this example is fine as a first step, let's merge this

Reply:
Actually sir we both have same thoughts but i am unable to explain completely 😂